src/cargo/core/shell.rs

```diff
@@ -484,16 +513,17 @@ mod imp {
                 // resize the console correctly, but there's no reasonable way
                 // to detect which kind of terminal we are running in, or if
                 // GetConsoleScreenBufferInfo returns accurate information.
-                return Some(cmp::min(60, width));
+                return TtyWidth::Known(cmp::min(60, width));
```

Comment:
Sorry if what I outlined was a little confusing. I think this (and only this) case should be Unknown, and Unknown should contain this `min(60, width)` value, so, maybe a better variant name would be Guess? or Uncertain?. `Unknown` is also fine.  Naming things is hard.

All the other usage of `Unknown` should be changed to `NoTty`.  IOW, if the computation ever fails, assume it isn't a tty.

Reply:
Ah, thanks! This should be fixed now.